tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -90,13 +90,32 @@ private static void LogMatchesIfDebugEnabled(IReadOnlyCollection<object>? result
             if (additiveContext != null)
             {
                 // run the WAF and execute the results
-                result = additiveContext.Run(args, _security.Settings.WafTimeoutMicroSeconds);
+                if (runWithEphemeral)
+                {
+                    result = additiveContext.RunWithEphemeral(args, _security.Settings.WafTimeoutMicroSeconds);
+                }
+                else
+                {
+                    result = additiveContext.Run(args, _security.Settings.WafTimeoutMicroSeconds);
+                }
+
                 RecordTelemetry(result);
+
+                // Show the deserilized string of additiveContext
+                var json = JsonConvert.SerializeObject(additiveContext);
+                Log.Debug("Additive context: {AdditiveContext}", json);
             }
         }
         catch (Exception ex) when (ex is not BlockException)
         {
-            Log.Error(ex, "Call into the security module failed");
+            if (logException is not null)
```

Comment:
Unclear to me what the benefit of passing the logger in rather than using the normal logger?

Reply:
It's a way to log the RASP information if something fails. It's used in RaspModule. The benefits are that we can distinguish between regular ASM calls and RASP calls if an error occurs, printing in this case the RASP arguments that caused the fail.